build.cake

```diff
@@ -203,17 +220,41 @@ Task("PackVSTemplates")
     .WithCriteria(() => IsRunningOnWindows())
     .Does(() =>
 {
-    var vsdirs = GetDirectories("./Templates/VisualStudio20*");
-    foreach (var vsdir in vsdirs)
-    {
-        DeleteFiles(vsdir.CombineWithFilePath("*.zip").FullPath);
-        var projdirs = GetDirectories(vsdir.CombineWithFilePath("*").FullPath);
-        foreach (var projdir in projdirs)
-        {
-            var outputPath = vsdir.CombineWithFilePath(projdir.GetDirectoryName() + ".zip");
-                Zip(projdir, outputPath);
-        }
-    }
+    var dotnet = Context.Tools.Resolve("dotnet.exe");
+    if (StartProcess(dotnet, "tool restore") != 0)
+        throw new Exception("dotnet tool restore failed.");
+
+    var result = StartProcess(
+        dotnet,
+        "vstemplate " +
+       $"-s Artifacts/MonoGame.Templates.CSharp/Release/MonoGame.Templates.CSharp.{version}.nupkg " +
+       $"--vsix Artifacts/MonoGame.Templates.CSharp/MonoGame.Templates.CSharp.{version}.vsix " +
+        "@Templates/VisualStudio/settings.rsp");
+
+    if (result != 0)
+        throw new Exception("dotnet-vstemplate failed to create VSIX.");
+});
+Task("PublishVSTemplates")
+    .IsDependentOn("PackVSTemplates")
+    .WithCriteria(() => IsRunningOnWindows())
+    .Does(() =>
+{
+    if (!GetVsixPublisher(out var vsixPublisher))
+        throw new Exception("Did not find VsixPublisher.exe.");
+
+    var pat = EnvironmentVariable("AZURE_DEVOPS_MARKETPLACE_PAT");
```

Comment:
@tomspilman I have the Personal Access Token (PAT). It'll need to be set as an environment variable in the teamcity build settings.